beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -145,7 +142,7 @@ func (s *Service) validateBeaconBlockPubSub(ctx context.Context, pid peer.ID, ms
 	}
 
 	// Handle block when the parent is unknown.
-	if !s.cfg.DB.HasBlock(ctx, bytesutil.ToBytes32(blk.Block().ParentRoot())) {
+	if !s.cfg.DB.HasBlock(ctx, bytesutil.ToBytes32(blk.Block().ParentRoot())) && !s.cfg.Chain.HasInitSyncBlock(bytesutil.ToBytes32(blk.Block().ParentRoot())) {
```

Comment:
This seems not trivial

What's the rationale for adding `!s.cfg.Chain.HasInitSyncBlock(bytesutil.ToBytes32(blk.Block().ParentRoot()))` ?

If it's not a hard fork specific change, should we have a discussion around it? perhaps another PR?

Reply:
Lets remove it from this PR then, I can add it in another one. Without this change a node can get permanently stuck when finally processing blocks after syncing an extended chunk of non finalized blocks.